### PR TITLE
Enhance blue-green deployment

### PR DESCRIPTION
**Pull request by ttsokov**
#### Description: 
1. Create getters/setters for MtaColor context variables.
    New context vars are created: VAR_DEPLOYED_MTA_COLOR and VAR_MTA_COLOR.
    JIRA: LMCROSSITXSADEPLOY-1200

2. Builders create default Metadata. Fix unit tests.
    The flow for creation of MTA model objects is Parser->Builder.
    Previously only the Parser was creating default Metadata
    for MTA objects, which leads to NPE when use only Builder
    to create MTA object. Now also Builders are creating default
    Metadata.
    JIRA: NGPBUG-68974

3. Refactor BlueGreenRenameStep dependency injection to be annotation based.
    JIRA: LMCROSSITXSADEPLOY-1200

### CODE REVIEW

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java [hunk 1]
```diff
@@ -176,10 +177,36 @@ static void setModuleContent(DelegateExecution context, String moduleName, byte[
...
+    public static ApplicationColor getDeployedMtaColor(DelegateExecution context) {
+        byte[] deployedMtaColorBinaryJson = (byte[]) context.getVariable(Constants.VAR_DEPLOYED_MTA_COLOR);
+        if (deployedMtaColorBinaryJson == null) {
+            return null;
+        }
+        return JsonUtil.getFromBinaryJson(deployedMtaColorBinaryJson, ApplicationColor.class);
+    }
+
+    public static void setDeployedMtaColor(DelegateExecution context, ApplicationColor deployedMtaColor) {
+        byte[] deployedMtaColorBinaryJson = JsonUtil.getAsBinaryJson(deployedMtaColor);
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java]: Why do we serialize an enum as a Json binary string?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java [hunk 2]
```diff
@@ -176,10 +177,36 @@ static void setModuleContent(DelegateExecution context, String moduleName, byte[
...
+    public static ApplicationColor getMtaColor(DelegateExecution context) {
+        byte[] mtataColorBinaryJson = (byte[]) context.getVariable(Constants.VAR_MTA_COLOR);
+        if (mtataColorBinaryJson == null) {
+            return null;
+        }
+        return JsonUtil.getFromBinaryJson(mtataColorBinaryJson, ApplicationColor.class);
+    }
+
+    public static void setMtaColor(DelegateExecution context, ApplicationColor mtaColor) {
+        byte[] mtaColorBinaryJson = JsonUtil.getAsBinaryJson(mtaColor);
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java]: Same.

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/ProcessDescriptorStep.java [hunk 3]
```diff
@@ -32,7 +32,7 @@
 
 public class ProcessDescriptorStep extends SyncActivitiStep {
 
-    private SecureSerializationFacade secureSerializer = new SecureSerializationFacade();
+    protected SecureSerializationFacade secureSerializer = new SecureSerializationFacade();
```
[the remark this answers is not in the document]
Comment (ttsokov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/ProcessDescriptorStep.java]: It is used from a child class. SecureSerializationFacade doesn't have state, isn't it?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java [hunk 4]
```diff
@@ -667,7 +694,7 @@ public static void setUnresolvedDeploymentDescriptor(DelegateExecution context,
         context.setVariable(Constants.VAR_MTA_UNRESOLVED_DEPLOYMENT_DESCRIPTOR, toBinaryYaml(unresolvedDeploymentDescriptor));
     }
 
-    static void setDeploymentDescriptor(DelegateExecution context, DeploymentDescriptor deploymentDescriptor) {
+    public static void setDeploymentDescriptor(DelegateExecution context, DeploymentDescriptor deploymentDescriptor) {
```
[the remark this answers is not in the document]
Comment (ttsokov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java]: It is used from a child class.

**Pull request merged**